src/backend/distributed/planner/distributed_planner.c

```diff
@@ -2293,3 +2294,56 @@ HasUnresolvedExternParamsWalker(Node *expression, ParamListInfo boundParams)
 									  boundParams);
 	}
 }
+
+
+RTEListProperties *
+GetRTEListPropertiesForQuery(Query *query)
+{
+	List *rteList = ExtractRangeTableEntryList(query);
+	return GetRTEListProperties(rteList);
+}
+
+
+/*
+ * GetRTEListProperties returns RTEListProperties struct processing the given
+ * rangeTableList.
+ */
+static RTEListProperties *
+GetRTEListProperties(List *rangeTableList)
+{
+	RTEListProperties *rteListProperties = palloc0(sizeof(RTEListProperties));
+
+	RangeTblEntry *rangeTableEntry = NULL;
+	foreach_ptr(rangeTableEntry, rangeTableList)
+	{
+		if (!(rangeTableEntry->rtekind == RTE_RELATION &&
+			  rangeTableEntry->relkind == RELKIND_RELATION))
+		{
+			continue;
+		}
+
+		Oid relationId = rangeTableEntry->relid;
```

Comment:
Here you can possibly get the citus table cache entry with `LookupCitusTableCacheEntry` so that it doesn't error for postgres tables and you can do the following checks with `IsCitusTableTypeCacheEntry`. (Might need to extend `IsCitusTableTypeInternal` a bit). The motivation is to not do unnecessary lookups, however this will only have impact in performance critical codepaths.